### PR TITLE
MCA-987 Change config to to work with multiple topics

### PR DESCRIPTION
**Pull request by nikolauska**
With these changes the mca-monitoring now properly supports multiple topics, but this will cause breaking change.

See difference between original and new config (`alarm` and `snsTopic`):

### Old
```yaml
custom:
  default:
    lambda:
      Errors:
        enabled: true
        alarm:
          threshold: 1
          evaluationPeriods: 1
        metric:
          period:
            minutes: 5
          statistic: Sum
          unit: Count
  snsTopic:
    name: Topic for mca monitoring critical alarms
    id: mca-alerts-critical-dev
    endpoints: []
    emails: []
```
### New
```yaml
custom:
  default:
    lambda:
      Errors:
        enabled: true
        alarm:
          warming:
            threshold: 1
            evaluationPeriods: 1
          critical:
            threshold: 2
            evaluationPeriods: 1
        metric:
          period:
            minutes: 5
          statistic: Sum
          unit: Count
  snsTopic:
    critical:
      name: Topic for mca monitoring critical alarms
      id: mca-alerts-critical-dev
      endpoints: []
      emails: []
    warning:
      name: Topic for mca monitoring warning alarms
      id: mca-alerts-warning-dev
      endpoints: []
      emails: []
```

There is one more change in here as well. aws-cdk packages were moved as peer dependency so whoever uses the library will decide which version of aws-cdk is used.

### CODE REVIEW

File: src/stacks/baseNestedStack.ts [hunk 1]
```diff
@@ -33,30 +33,48 @@ export default class BaseNestedStack extends cfn.NestedStack {
...
     metricName: string,
-    localConf: config.ConfigMetricAlarms,
+    localConf: config.ConfigMetricAlarm,
     dimensions?: object,
   ): void {
-    const autoResolve = config.configAutoResolve(this.defaultType, metricName, localConf);
-    const isEnabled = config.configIsEnabled(this.defaultType, metricName, localConf);
-    const suffix = config.configAlarmSuffix(this.defaultType, metricName, localConf);
-    const fullMetricName = suffix ? `${metricName}-${suffix}` : metricName;
+    const isEnabled = localConf.enabled !== false;
```
Comment (michalksiezuk-nc) [inline src/stacks/baseNestedStack.ts]: Isn't `isEnabled` redundant?

Comment (michalksiezuk-nc) [reply src/stacks/baseNestedStack.ts]: It's being used only once in L41 and yet it's being checked for falsy value.

File: src/stacks/nestedAccount.ts [hunk 2]
```diff
@@ -25,20 +25,51 @@ export class NestedAccountAlarmsStack extends cfn.NestedStack {
   }
 
   private setupAccountAlarm(metricName: string): void {
-    const autoResolve = config.configAutoResolve(config.ConfigDefaultType.Account, metricName);
+    const conf = config.configGetAllDefaults(config.ConfigDefaultType.Account)?.[metricName];
+    if (!conf) {
+      return;
+    }
+
+    const isEnabled = conf.enabled;
```
Comment (michalksiezuk-nc) [inline src/stacks/nestedAccount.ts]: This const is redundant.

File: src/stacks/nestedLogGroup.ts [hunk 3]
```diff
@@ -63,17 +48,15 @@ export class NestedLogGroupAlarmsStack extends BaseNestedStack {
...
     pattern: string,
-    localConf: config.ConfigMetricAlarms,
+    localConf: config.ConfigMetricAlarm,
   ): void {
-    const isEnabled = config.configIsEnabled(this.defaultType, metricFilterName, localConf);
-    const suffix = config.configAlarmSuffix(this.defaultType, metricFilterName, localConf);
-    const fullMetricFilterName = suffix ? `${metricFilterName}-${suffix}` : metricFilterName;
-
-    if (!isEnabled) {
+    const isEnabled = localConf.enabled !== false;
```
Comment (michalksiezuk-nc) [inline src/stacks/nestedLogGroup.ts]: This const is redundant.

File: src/stacks/nestedLogGroup.ts [hunk 4]
```diff
@@ -63,17 +48,15 @@ export class NestedLogGroupAlarmsStack extends BaseNestedStack {
...
+    const localEnabled = Object.values(localConf.alarm || {}).find(l => l.enabled === true);
```
Comment (michalksiezuk-nc) [inline src/stacks/nestedLogGroup.ts]: This could be extracted as library code, it's being used in at least two different places.

**Comment by michalksiezuk-nc** [comment]
Really nice work Niko, just a few minor things I found that you could consider.

File: src/stacks/baseNestedStack.ts (hunk 1 above)
Comment (nikolauska) [reply src/stacks/baseNestedStack.ts]: We want to make sure that undefined is not false as by default it should be enabled

Comment (nikolauska) [reply src/stacks/baseNestedStack.ts]: This allows you to disable alerts for all topics or just single topic while still keeping all configs

File: src/stacks/nestedAccount.ts (hunk 2 above)
Comment (nikolauska) [reply src/stacks/nestedAccount.ts]: That is actually bug, it should check if the value is false so undefined is marked as enabled

File: src/stacks/baseNestedStack.ts (hunk 1 above)
Comment (michalksiezuk-nc) [reply src/stacks/baseNestedStack.ts]: What I meant is that it could be inlined since it's used only once

Comment (michalksiezuk-nc) [reply src/stacks/baseNestedStack.ts]: But I'll leave it to you, approving :slightly_smiling_face:

**Review by michalksiezuk-nc — APPROVED** [review]

File: src/stacks/nestedLogGroup.ts (hunk 4 above)
Comment (nikolauska) [reply src/stacks/nestedLogGroup.ts]: Will do actually

**Review by michalksiezuk-nc — APPROVED** [review]

**Pull request merged**